srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The steps needed to use a custom type in *RTI Perftest* are:
+
+Copy the IDL files into `~/rtiperftest/srcIdl/customTypes/`
+-----------------------------------------------------------
+
+There are some restrictions for this step:
+
+- You can include multiple `.idl` files, but all of them should be at the same
+  level (no folders are allowed inside the `customTypes` folder.
+- The `idl` files cannot be named `custom.idl`. A file with this name will be generated
+  automatically by the `build` script, so this file would be overwritten.
+- The main type you will link should be declared as ``struct <NameOfTheType> {``.
+  The `struct` and `{` words should be spaced as shows. This way the build scripts
+  will be able to find the type accross a list of `.idl`.
+
+Implement the API custom type functions of `customType.cxx`
+-----------------------------------------------------------
+
+You should find a file in `${PERFTEST_HOME}/srcCpp` named `customType.cxx`. This
+file contains several functions that must be implemented in order to correctly
+initialize, finalize and populate the sample.
+
+**Note:** *RTI Perftest* will not initialize by default sequences, optional members
+or non-primitive structures. This means that those fields will need to be
+initialized by the user in the `set_custom_type_data`, `set_custom_type_dynamic_data`
+and `set_custom_type_data_flatdata`.
+
+**Note:** If using *FlatData* see the dedicated section below.
+
+Run the `build` script adding ``--customType <NameOfTheType``
+-------------------------------------------------------------
+
+Run the ``build.sh`` or ``build.bat`` script using the ``--customType`` option.
+
+**Note:** If using *FlatData* see the dedicated section below.
+
+Run *RTI Perftest* as usual
+---------------------------
+
+A string at the beginning of the execution should indicate that you are using a
+custom type and that this is not a regular execution.
+
+
+Custom Types + `FlatData`
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+*FlatData* types are at this point in *RTI Perftest* associated to regular types.
+If you want to use *FlatData* with for your Custom Types, follow these
+additional steps:
+
+- Create a copy of your original type (which should not be `FlatData`) in the same IDL file.
+- Make this new type mutable (``@mutable``) and make it FlatData compatible (``@language_binding(FLAT_DATA)``).
+- Implement the additional API custom type functions of customtype.cxx for *FlatData* types.
+  This is a *must*, as the type is ``@mutable``, and every field is considered as optional
+  and not sent into the wire if it is not set.
+- Run the build script with the command-line parameter ``--customType <type>`` and ``--customTypeFlatData <flat_type>``.
+- Run *RTI Perftest* as usual.
+
+Full example using Custom Types
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following types will be used for this example. The code can be found under
+``resource/examples_customType/no_key_large``.
```

Comment:
In my opinion, this isn't a great example to document...it's not really a "Full example".  It's not a keyed datatype nor does it show flat data.